als-suggestions/shared/src/main/scala/org/mulesoft/als/suggestions/styler/JsonSuggestionStyler.scala

```diff
@@ -23,6 +26,15 @@ case class JsonSuggestionStyler(override val params: SyamlStylerParams) extends
   override protected def renderYPart(part: YPart, indentation: Option[Int] = None): String =
     JsonRender.render(part, indentation = indentation.getOrElse(0), buildRenderOptions)
 
+  /**
+    * This method is used to escape the backslash character for non snippets.
+    * @param fixedJson rendered and fixed suggestion
+    * @param newText the original suggestion
+    * @return the suggestion escaped
+    */
+  private def escapeNonSnippets(fixedJson: String, newText: String): String =
+    fixedJson.replace(newText, "\\".concat(newText))
```

Comment:
what if the $ is not the first char in the snippet?

Reply:
(in case it must always be the first char, then the "if contains" in `rawRender` should be a `startsWith`?, maybe add a test in which the $ is not at the start?)

Reply:
@llibarona this code block was change